sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -793,8 +800,18 @@ export class MessageReceiver extends LinkEntity {
         const config: RetryConfig<void> = {
           operation: () =>
             this._init(options).then(() => {
-              if (this._receiver && this.receiverType === ReceiverType.streaming) {
-                this._receiver.addCredit(this.maxConcurrentCalls);
+              if (this.wasCloseInitiated) {
+                log.error(
+                  "[%s] close() method of Receiver '%s' with address '%s' was called. " +
+                    "by the time the receiver finished getting created. Hence, disallowing messages from being received. ",
+                  connectionId,
+                  this.name,
+                  this.address
```

Comment:
At this point we need to close the receiver as well.

Reply:
`wasCloseInitiated` indicates that close is invoked and in progress right. Why call close() again?

Reply:
Take the case of user calling `close()` when `init()` was in progress which by the way is the case for which #2199 was created based on #1730

When `close()` runs, it won't see any open receiver (because it got closed due to network loss)

When `init()` completes, we now have an open receiver

Reply:
Ok, updated - this seems to be one possible scenario depending on timing of close(), if it was called during and not after init() completes. 

Overall, introduction of flags (wasCloseInitiated, isItselfClosed, isConnecting, isOpen, etc) to just track state of receiver seems to indicate a bigger design problem to me. And maybe a more detailed discussion is needed about the event loop to be on same page about validation of these scenarios. 

@ramya-rao-a